src/App.jsx

```diff
@@ -14,33 +18,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_ALPHABETICALLY = 'Sort alphabetically';
+const SORT_BY_LENGTH = 'Sort by length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goods, { sortField, reversed }) {
+  let preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      if (sortField === SORT_ALPHABETICALLY) {
+        return good1.localeCompare(good2);
+      }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+      if (sortField === SORT_BY_LENGTH) {
+        return good1.length - good2.length;
+      }
+
+      return 0;
+    });
+  }
+
+  if (reversed) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    reversed,
+  });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const resetSorting = () => {
+    setSortField('');
+    setReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_ALPHABETICALLY)}
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SORT_ALPHABETICALLY,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_BY_LENGTH)}
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SORT_BY_LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setReversed(!reversed)}
+          className={cn('button', 'is-warning', {
+            'is-light': !reversed,
+          })}
+        >
+          Reverse
+        </button>
+
+        {sortField || reversed ? (
+          <button
+            type="button"
+            onClick={resetSorting}
+            className="button is-danger"
+          >
+            Reset
+          </button>
+        ) : null}
```

Comment:
You don't need the ternary operator if you return nothing in the second case, use `&&` instead